### DISCUSSION TITLE
Caps lock notification within Standard Notes unlock

### DISCUSSION THREAD

**Discussion by saphirahopeseeker** [standardnotes/forum#2639] [Suggestions]
When you're unlocking there is no notification of if caplock is set, leading to possible frustration. 

There are a few possibilities that I can think of to address this:
1) If the previous passcode == current passcode ask them if they meant to have caps lock on 
2) Show a little caplock icon
3) Do nothing. As Zedgoat and ghewgill noted there is a light on most keyboard and it feeds into the 'caps lock is an anti pattern'

Furthermore caplocks maybe leaks information about the type of password to anyone watching your screen so maybe it's a bad idea to actually display it prominently like SN used to with the 'Protected'.

**Comment by JaspalSuri** [comment]
Hi @standardorcawolf, thank you for creating a thread here for the suggestion that was shared on [Slack](https://standardnotes.org/slack)! I'll be sure to take note of this as it appears to be a good suggestion and it's been used in other applications. 

I don't think 1 would be possible since [(IIRC)](https://standardnotes.org/help/79/how-does-standard-notes-encrypt-data-on-my-device) the apps do not store the passcode keys.

**Comment by saphirahopeseeker** [comment]
It would be if the following occured:

1) When passcode is entered we cache it until the application unlocks.
2) In the event the passcode is incorrect, we reset input and wait for
second input.
3) Before decrypting again we compare if this current entry is equal to
last entry and hint that they should check capslock. (We then flush the
cached passcode).

This is similar to what Veracrypt does for the preboot authentication
environment. In the event there are 5 failed attempts it shows a long
message indicating that either 'boot header is damanged or something
along the lines of caplocks and such'.
